org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/javadoc/JavaDocHTMLPathHandler.java

```diff
@@ -0,0 +1,355 @@
+			return text;
+		}
+
+		String[] names = fragmentName.split("\\.");
+		String fragmentPath = "";
+		for (String name : names) {
+			fragmentPath += name + "/";
+		}
+
+		srcPath = srcPath.replace('\\', '/');
+
+		String relativeToJarPath = fragmentPath + srcPath; // /relative/path/to/file.abc
+		InputStream is = null; // file from jar (ZipEntry)
+		JarFile jar = null;
+
+		try {
+			File currentJarPath = null;
+
+			ZipEntry currentZipEntry = null;
+
+			URL javadocJarBaseLocationURL = JavaDocLocations.getJavadocBaseLocation(internalJarFragment); //Absolute location of javadoc jar (not class or source jar)
+			//Attempt to get file from javadoc jar
+			if (javadocJarBaseLocationURL != null) {
+				URI javadocJarBaseLocationURI = javadocJarBaseLocationURL.toURI();
+				currentJarPath = getJarPathFromURI(javadocJarBaseLocationURI);
+				if (currentJarPath != null) {
+					jar = new JarFile(currentJarPath);
+					currentZipEntry = jar.getEntry(relativeToJarPath);
+				}
+			}
+
+			//No file was in the javadoc jar, try the source jar
+			if (currentZipEntry == null) {
+				currentJarPath = SourceJarLocations.getSourceJarPath(internalJarFragment); //Absolute location of source jar
+				if (currentJarPath != null) {
+					jar = new JarFile(currentJarPath);
+					currentZipEntry = jar.getEntry(relativeToJarPath);
+				}
+			}
+
+			if (jar == null || currentZipEntry == null) {
+				return text; //File from source path could not be located in either jar
+			}
+
+
+
+			String jarRootName = internalJarFragment.getAncestor(IJavaElement.PACKAGE_FRAGMENT_ROOT).getElementName();
+			if (jarRootName.endsWith(".jar")) {
+				jarRootName = jarRootName.substring(0, jarRootName.length() - 4);
+			}
+			jarRootName += "/";
+
+			String outputPath = "file:" + EXTRACTED_JAR_IMAGES_FOLDER + jarRootName + fileName;
+			URI outputURI = new URI(outputPath);
+			File outputFile = new File(outputURI.getPath());
+
+			//Check if the file actually needs to be extracted
+			if (!outputFile.exists()) {
+				is = jar.getInputStream(currentZipEntry);
+				extractFileTo(is, outputFile);
+			} else {//Check if the file is outdated
+				BasicFileAttributes existingOutputFileAttributes = Files.readAttributes(outputFile.toPath(), BasicFileAttributes.class);
+				FileTime existingFileCreationTime = existingOutputFileAttributes.creationTime();
+				BasicFileAttributes jarOutputFileAttributes = Files.readAttributes(currentJarPath.toPath(), BasicFileAttributes.class);
+
+				FileTime jarFileCreationTime = jarOutputFileAttributes.creationTime();
+
+				if (jarFileCreationTime.compareTo(existingFileCreationTime) > 0) {
+					is = jar.getInputStream(currentZipEntry);
+					extractFileTo(is, outputFile);
+				}
+			}
+
+			//Insert new path into text
+			return text.substring(0, offsets[0]) + outputPath + text.substring(offsets[1]);
+
+		} catch (JavaModelException e3) {
+			return text;
+		} catch (IOException e) {
+			return text;
+		} catch (URISyntaxException e) {
```

Comment:
Fails on my mac with: java.net.URISyntaxException: Illegal character in path at index 39: file:/Users/fbricon/Library/Application Support/Code - Insiders/User/workspaceStorage/37a4ce5c535ed97b5844be5134041ff1/redhat.java/jdt_ws/.metadata/.plugins/org.eclipse.jdt.ls.core/extracted-jar-images/reactor-core-3.2.9.RELEASE/error.svg

You need to make sure all URIs you create handle white space properly